compiler/rustc_data_structures/src/remap.rs

```diff
@@ -0,0 +1,28 @@
+/// Remaps the type with a different lifetime for 'tcx if applicable.
+pub trait Remap {
```

Comment:
There is `rustc_middle::ty::parametrized::ParameterizedOverTcx` that does the same thing.